app/src/main/java/com/cs102/studybuddy/Register.java

```diff
@@ -0,0 +1,103 @@
+package com.cs102.studybuddy;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+
+public class Register extends AppCompatActivity {
+
+    private EditText regEmail, regPass, regConfirm, year, gender;
+    private Button signUp;
+    private FirebaseAuth mAuth;
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Check if user is signed in (non-null) and update UI accordingly.
+//        FirebaseUser currentUser = mAuth.getCurrentUser();
+//        if (currentUser != null){
+//            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+//            startActivity(intent);
+//            finish();
+//        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        mAuth = FirebaseAuth.getInstance();
+
+        regEmail = findViewById(R.id.email_signup);
+        regPass = findViewById(R.id.password_signup);
+        regConfirm = findViewById(R.id.confirm_password_signup);
+        year = findViewById(R.id.year_birth);
+        gender = findViewById(R.id.gender);
+        signUp = findViewById(R.id.SignUp);
+
+        signUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String email, pass, passconfirm;
+                email = String.valueOf(regEmail.getText());
+                pass = String.valueOf(regPass.getText());
+                passconfirm = String.valueOf(regConfirm.getText());
+
+                if (TextUtils.isEmpty(email)){
+                    Toast.makeText(Register.this, "Enter Email", Toast.LENGTH_LONG).show();
+                    return;
+                }
+
+               if (!email.endsWith("bilkent.edu.tr")) {
+                    Toast.makeText(Register.this, email.substring(email.length()-14), Toast.LENGTH_LONG).show();
```

Comment:
This is causing `OutOfBoundsException` when given an email that is less than 14 characters. A possible message would be:

```suggestion
                    Toast.makeText(Register.this, "Email must be a Bilkent email", Toast.LENGTH_LONG).show();
```